src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -50,26 +50,9 @@ Result Swapchain::Create(const grfx::SwapchainCreateInfo* pCreateInfo)
 
     // Create color images if needed. This is only needed if we're creating
     // a headless swapchain.
-    if (mColorImages.empty()) {
-        for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
-            grfx::ImageCreateInfo rtCreateInfo = ImageCreateInfo::RenderTarget2D(mCreateInfo.width, mCreateInfo.height, mCreateInfo.colorFormat);
-            rtCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
-            rtCreateInfo.RTVClearValue         = {0.0f, 0.0f, 0.0f, 0.0f};
-            rtCreateInfo.initialState          = grfx::RESOURCE_STATE_PRESENT;
-            rtCreateInfo.usageFlags =
-                grfx::IMAGE_USAGE_COLOR_ATTACHMENT |
-                grfx::IMAGE_USAGE_TRANSFER_SRC |
-                grfx::IMAGE_USAGE_TRANSFER_DST |
-                grfx::IMAGE_USAGE_SAMPLED;
-
-            grfx::ImagePtr renderTarget;
-            ppxres = GetDevice()->CreateImage(&rtCreateInfo, &renderTarget);
-            if (Failed(ppxres)) {
-                return ppxres;
-            }
-
-            mColorImages.push_back(renderTarget);
-        }
+    ppxres = CreateColorImages();
```

Comment:
nit: Maybe we should add a check to do this only if the swapchain is headless? Otherwise it has to happen after the API specific create, it probably always will, but maybe better to make that explicit.
Also wouldn't matter when we split the swapchain classes.

Reply:
If API specific create failed, then it won't get to this part, otherwise this behave the same as `CreateDepthImages`, create image if it's not exist.